src/bokeh/core/property/instance.py

```diff
@@ -111,6 +114,9 @@ def instance_type(self) -> type[T]:
 
     def validate(self, value: Any, detail: bool = True) -> None:
         super().validate(value, detail)
+        # Try converting to Narwhals so that any dataframe supported by Narwhals
+        # can be supported by ColumnDataSource.
+        value = nw.from_native(value, pass_through=True, allow_series=True)
```

Comment:
@MarcoGorelli I don't think it makes sense for `Object` to have anything like this inside, it is intended to be completely generic. My suggestion with adding the other new dataframe property types was so that hopefully we would not need `Object("narwhals.Series")` etc anywhere. Does that seem feasible?

Reply:
thanks - I'm a bit short on time right now, would you have time to add a commit to show how you'd like it done? if not I can pick this up again next year, no worries